src/ConfigFactory.Avalonia/Builders/NumericControlBuilder.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Avalonia.Controls;
+using Avalonia.Data;
+using Avalonia.Layout;
+using ConfigFactory.Avalonia.Helpers;
+using ConfigFactory.Core;
+using ConfigFactory.Core.Attributes;
+using ConfigFactory.Generics;
+using System.Reflection;
+
+namespace ConfigFactory.Avalonia.Builders;
+
+public class NumericControlBuilder : ControlBuilder<NumericControlBuilder>
+{
+    public override object? Build(IConfigModule context, PropertyInfo propertyInfo)
+    {
+        // build as drop down
+        if (propertyInfo.GetCustomAttribute<DropdownConfigAttribute>() is DropdownConfigAttribute dca)
+        {
+            return DropdownBuilder.Build(context, propertyInfo, dca);
+        }
+
+        // build as numeric control
+        var control = new NumericUpDown
+        {
+            DataContext = context,
+            VerticalAlignment = VerticalAlignment.Top,
+            [!NumericUpDown.ValueProperty] = new Binding(propertyInfo.Name)
+        };
+
+        if (propertyInfo.GetCustomAttribute<NumericConfigAttribute>() is NumericConfigAttribute nca)
+        {
+            if (TryConvertToDecimal(nca.Minimum, out decimal min))
+            {
+                control.Minimum = min;
+            }
+            if (TryConvertToDecimal(nca.Maximum, out decimal max))
+            {
+                control.Maximum = max;
+            }
+            if (TryConvertToDecimal(nca.Increment, out decimal inc))
+            {
+                control.Increment = inc;
+            }
```

Comment:
In regards to the last comment, the decimal checks aren't required:

(At the very least use `decimal?` and pattern match for decimal)

Reply:
As I recall, if you set attribute property types to `decimal?`, you will also have to set attribute properties explicitly (like `Increment = 1.0m`).
I guess it is not a convinient way.

Reply:
Nah, that's only if it's not nullable, and even then you don't need to explicitly assign them, it'll just throw a warning if it's undefined. (The revision I made sets the default value, so there won't be warnings and the caller won't need to assign them)

Reply:
Can you try it yourself?

I am getting compilation errors after simply changing types to `decimal?`:

```
'Minimum' is not a valid named attribute argument because it is not a valid attribute parameter type ConfigFactory.Avalonia.Demo	
```

Reply:
Ah my bad, forgot that decimal was a runtime type. You could use `double`/`double?` though, and then just cast to `decimal` at runtime.

Reply:
Then I would prefer to keep a generic `object?` in case when someone wants to use big integers.

Reply:
Ah fair enough, guess that'll have to do 👍